tests/unit/test_execution_plan_clone.cpp

```diff
@@ -0,0 +1,416 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "gtest.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdio.h>
+#include <string.h>
+#include "../../src/query_ctx.h"
+#include "../../src/util/rmalloc.h"
+#include "../../src/util/arr.h"
+#include "../../src/execution_plan/execution_plan.h"
+#include "../../src/arithmetic/agg_funcs.h"
+#include "../../src/arithmetic/funcs.h"
+#include "../../src/procedures/procedure.h"
+
+#ifdef __cplusplus
+}
+#endif
+
+class ExecutionPlanCloneTest: public ::testing::Test {
+  protected:
+	static void SetUpTestCase() {
+		// Use the malloc family for allocations
+		Alloc_Reset();
+		// Init query context.
+		ASSERT_TRUE(QueryCtx_Init());
+		// Initialize GraphBLAS.
+		GrB_init(GrB_NONBLOCKING);
+		GxB_Global_Option_set(GxB_FORMAT, GxB_BY_COL); // all matrices in CSC format
+		GxB_Global_Option_set(GxB_HYPER, GxB_NEVER_HYPER); // matrices are never hypersparse
+		Proc_Register();         // Register procedures.
+		AR_RegisterFuncs();      // Register arithmetic functions.
+		Agg_RegisterFuncs();     // Register aggregation functions.
+
+		// Create a graphcontext
+		_fake_graph_context();
+	}
+
+	static void parse_query(const char *query, AST **ast, ExecutionPlan **plan) {
+
+		cypher_parse_result_t *parse_result = cypher_parse(query, NULL, NULL, CYPHER_PARSE_ONLY_STATEMENTS);
+		*ast = AST_Build(parse_result);
+		*plan = NewExecutionPlan();
+	}
+
+	static void _fake_graph_context() {
+		GraphContext *gc = (GraphContext *)malloc(sizeof(GraphContext));
+
+		gc->g = Graph_New(16, 16);
+		gc->index_count = 0;
+		gc->graph_name = strdup("G");
+		gc->attributes = raxNew();
+		gc->string_mapping = (char **)array_new(char *, 64);
+		gc->node_schemas = (Schema **)array_new(Schema *, GRAPH_DEFAULT_LABEL_CAP);
+		gc->relation_schemas = (Schema **)array_new(Schema *, GRAPH_DEFAULT_RELATION_TYPE_CAP);
+		QueryCtx_SetGraphCtx(gc);
+	}
+
+	static void ExecutionPlan_OpsEqual(const ExecutionPlan *plan_a, const ExecutionPlan *plan_b,
```

Comment:
I don't think this is a thing to do right now since
1. this is misleading. They are structural equivalent but may be logical (filter tree/arithmetic expressions) not equal.
2. used only in testing for now.

Unless we will have a full `Equal` method, we should keep it only in the test file IMO